### ISSUE TITLE
Custom markers

### ISSUE THREAD

**Issue by 2dvisio** [ayastreb/jekyll-maps#30]
The Google Maps API provides the option of customising the marker image, see here:

https://developers.google.com/maps/documentation/javascript/custom-markers

Is it possible to extend the Location to include a `marker_icon` so that one could set a custom image.

**Comment by ayastreb** [comment]
Should be possible, PRs welcome!
I'm not sure if I'll have time in the nearest future for this.
I'll try to tackle it once I get some free time, if it's not already implemented by someone else by then 😸

**Comment by schaechinger** [comment]
I think the problem is that google.maps.Marker requires a `icon` property to work.
The property is called `image` here if I am not mistaken.

**Comment by 2dvisio** [comment]
@schaechinger I thought that image was something else, but I see now that you are right. Anyone who needs to customise the marker should use image.
Thank you!

**Comment by 2dvisio** [comment]
I have re-opened the issue as I have tested the `image` property, but that only assign an image to be opened after you click the marker. Unfortunately it does not change the marker itself.
Thank you in any case!

**Comment by schaechinger** [comment]
You're right, the property 'icon' is the one that is used by the marker class but is not implemented, yet

**Comment by ayastreb** [comment]
Right, marker icon is not implemented yet and it always shows default.

How would you like to the set custom marker icon?
- Globally, e.g. in `config.yml`
- per map, e.g. `{% google_map marker_icon="https://maps.google.com/mapfiles/kml/shapes/info-i_maps.png" %}`
- per location, e.g. 
```
# _posts/2016-01-01-post.md
location:
  latitude: 51.5285582
  longitude: -0.2416807
  marker_icon: "https://maps.google.com/mapfiles/kml/shapes/info-i_maps.png"
```

**Comment by 2dvisio** [comment]
Ideally this should follow the Google API. So I would say per location would make more sense? What do you think?

**Comment by ayastreb** [comment]
I think it makes sense both in location and as map attribute :) and it shouldn't be very difficult to implement. I'll try to do it in the next couple of days.

**Comment by ayastreb** [comment]
Implemented in 2.2.0!
Check out this example - https://ayastreb.me/jekyll-maps/#examples-custom-marker
You can set custom marker for the whole map with `marker_icon` attribute on the map tag, or you can set a custom marker for specific location with `marker_icon` key for that location.

**Comment by 2dvisio** [comment]
I think it is done! Thank you very much for your responsiveness! 
🖖👍
